src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlDependencyListener.cs

```diff
@@ -100,7 +101,15 @@ internal SqlConnectionContainer(SqlConnectionContainerHashHelper hashHelper, str
                     CommandText = "select is_broker_enabled from sys.databases where database_id=db_id()"
                 };
 
-                if (!(bool)_com.ExecuteScalar())
+                // db_id() returns the database ID of the current database hence it will always be one line result
+                using SqlDataReader reader = _com.ExecuteReader(CommandBehavior.SingleRow);
+                if (reader.Read() && reader[0] is not null)
+                {
+                    dbId = reader.GetBoolean(0);
+                }
+                reader.Close();
```

Comment:
Can we using bracketed `using` for `reader`? You wouldn't need to call this in that case.

Reply:
In general I'm not a fan of the new unscoped using syntax. I can understand that in quick projects it has some value but for highly formal code like a base library I think it's better to be explicit, what do you think?

Reply:
Agreed. I will fix it in the next commit.